### DISCUSSION TITLE
Coralai M.2, USB and "automatic throttling"?

### DISCUSSION THREAD

**Discussion by Antoine-H** [blakeblackshear/frigate#3110] [Ask A Question] [answered]
Hello,

I hope here is the correct place for such support question.

I can't decide if I should get an M.2, PCIe, or USB coralai. The USB seems easier because it's plug and play, no drivers apparently, but the documentation says that it lacks "automatic throttling", which I have no idea what it means.

Could someone advise me, and explain what automatic throttling is?

My server runs Debian and it currently has two RTMP cameras (Reolink) attached to it.

Thanks a lot

**Comment by NickM-27** [comment, marked as answer]
Throttling will help protect the device from overheating, making it potentially last longer in the long term. 

Overall, I believe frigate runs the coral at max clock, but not constantly (generally, depends on how much motion your cameras see).

The way things are right now though, if you can find a coral at all you're doing pretty well.

**Reply by NickM-27** [reply]
Here's a git issue with more context: https://github.com/google-coral/edgetpu/issues/150

**Reply by Antoine-H** [reply]
Ah throttling means running slower, ok thanks :).

Ok I see, so it would be best to get a M.2 or PCIe one if I can, this way it doesn't run at max clock when it doesn't need to and it would benefit from the fans inside my server.

Aaand, I wasn't aware there was a shortage of it..

**Reply by NickM-27** [reply]
> Ah throttling means running slower, ok thanks :).
> 
> Ok I see, so it would be best to get a M.2 or PCIe one if I can, this way it doesn't run at max clock when it doesn't need to and it would benefit from the fans inside my server.
> 
> Aaand, I wasn't aware there was a shortage of it..

One point of clarification, it will always run at max clock when it's not hotter than the throttling temp.
